### PR TITLE
deprecate histogram functionality

### PR DESCRIPTION
**Pull request by simonbyrne**
Updated PR from #6842

### CODE REVIEW

**Comment by tkelman** [comment]
I think the `export`s for these should probably also be moved from `exports.jl` to near their definitions in `deprecated.jl`

File: test/random.jl [hunk 1]
```diff
@@ -334,17 +334,23 @@ for rng in ([], [MersenneTwister()], [RandomDevice()])
     end
 end
 
+function hist(X,n)
+    v = zero(Int,n)
```
Comment (tkelman) [inline test/random.jl]: `zeros` ?

Comment (simonbyrne) [inline test/random.jl]: Ah, right.

**Comment by tkelman** [comment]
@simonbyrne this wasn't added to NEWS, was it?

**Comment by simonbyrne** [comment]
No, I don't think so; the change looks good.

File: base/deprecated.jl [hunk 2]
```diff
@@ -1273,6 +1273,164 @@ end
...
+        end
+    end
+    start = step*(ceil(lo/step)-1)
+    nm1 = ceil(Int,(hi - start)/step)
+    start:step:(start + nm1*step)
+end
+
+## midpoints of intervals
+midpoints(r::Range) = r[1:length(r)-1] + 0.5*step(r)
+midpoints(v::AbstractVector) = [0.5*(v[i] + v[i+1]) for i in 1:length(v)-1]
```
Comment (tkelman) [inline base/deprecated.jl]: there were missing depwarns - they're still exported and tested...

**Pull request merged**